### PR TITLE
conf: 'cylc' => 'scheduler'

### PR DESCRIPTION
**Pull request by wxtim**
These changes partially address #3696

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.

Change log and documentation to be consilidated for this change as part of #3696 .

#### Blow by blow account of changes.
- Moved config items in `[cylc]` to `[scheduler]`.
- Remove deprecated items `[documentation]` and `[document viewers]`
- Test `functional/modes/00-simulation.t` on inspection appeared to be testing force-run-mode (despite the name) - I don't understand how it got through the review of that PR.

I could do other changes (Moving suite servers into scheduler item, for example), but I've decided that this is sufficient and complete as a PR and the change from cylc => scheduler is going to be a source of pain if this PR is left open.

### CODE REVIEW

File: cylc/flow/cfgspec/globalcfg.py [hunk 1]
```diff
@@ -94,9 +94,9 @@
         The number of old run directory trees to retain at start-up.
     ''')
 
-    # suite
-    with Conf('cylc', desc='''
-        Default values for entries in flow.cylc ``[cylc]`` section.
+    with Conf('scheduler', desc='''
+        Default values for entries in flow.cylc ``[scheduler]`` section.
+        This should not be confused with scheduling in the ``flow.conf`` file.
```
Comment (datamel) [inline cylc/flow/cfgspec/globalcfg.py:99]: Does this make it sound as if our config names are confusing?

Comment (datamel) [reply cylc/flow/cfgspec/globalcfg.py:99]: Also, should it be...? 
```suggestion
        This should not be confused with scheduling in the ``flow.cylc`` file.
```

File: tests/flakyfunctional/cylc-take-checkpoints/00-basic/flow.cylc [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!jinja2
```
Comment (oliver-sanders) [inline tests/flakyfunctional/cylc-take-checkpoints/00-basic/flow.cylc:1]: Methinks this is a rebase error.

Comment (wxtim) [reply tests/flakyfunctional/cylc-take-checkpoints/00-basic/flow.cylc:1]:
```suggestion
```

File: cylc/flow/scheduler.py [hunk 3]
```diff
@@ -1094,8 +1096,8 @@ def load_flow_file(self, is_reload=False):
             share_dir=self.suite_share_dir,
         )
         self.cylc_config = DictTree(
-            self.config.cfg['cylc'],
-            glbl_cfg().get(['cylc'])
+            self.config.cfg['scheduler'],
+            glbl_cfg().get(['scheduler'])
```
[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/flow/scheduler.py:1100]: I came across it before - was there anything that you wanted me to do about it?

**Comment by datamel** [comment]
I'm getting a fail on `tests/f/events/15-host-task-event-handler-retry-globalcfg.t`. Although this appears to be broken on master - probably from https://github.com/cylc/cylc-flow/pull/3908.

**Comment by datamel** [comment]
> I'm getting a fail on `tests/f/events/15-host-task-event-handler-retry-globalcfg.t`. Although this appears to be broken on master - probably from #3908.

Ah, I see @oliver-sanders is deleting it in https://github.com/cylc/cylc-flow/pull/3899.

File: cylc/flow/cfgspec/globalcfg.py [hunk 4]
```diff
@@ -94,9 +94,10 @@
...
-    # suite
-    with Conf('cylc', desc='''
-        Default values for entries in flow.cylc ``[cylc]`` section.
+    with Conf('scheduler', desc='''
+        Default values for entries in :cylc:conf:`flow.cylc[scheduler]`
+        section. This should not be confused with scheduling in the
+        ``flow.cylc`` file.
     '''):
         Conf('UTC mode', VDR.V_BOOLEAN, False, desc='''
                 Default for :cylc:conf:`flow.cylc[cylc]UTC mode`.
```
Comment (datamel) [inline cylc/flow/cfgspec/globalcfg.py:103]:
```suggestion
                Default for :cylc:conf:`flow.cylc[scheduler]UTC mode`.
```
A search for [cylc] shows a few of these remaining. e.g in `suite.py`.

**Review by datamel — CHANGES_REQUESTED** [review]
Tim, I have run the test battery - 2 failing tests which are not specific to this branch.
There do remain some references to [cylc] which breaks the docs build.
Once these are fixed. Looks great!

**Review by datamel — APPROVED** [review]
Great, feedback addressed. Thanks Tim.

**Comment by wxtim** [comment]
@oliver-sanders  - Can I prod your re-review

File: cylc/flow/scheduler.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply cylc/flow/scheduler.py:1088]: Nope, just something to keep somewhere in the back of our minds for future support queries.

File: tests/flakyfunctional/cylc-take-checkpoints/00-basic/flow.cylc [hunk 5]
```diff
@@ -0,0 +1,25 @@
+[scheduler]
```
Comment (oliver-sanders) [inline tests/flakyfunctional/cylc-take-checkpoints/00-basic/flow.cylc:1]: What's this file doing here? Rebase error?

File: tests/flakyfunctional/cylc-take-checkpoints/00-basic/flow.cylc (hunk 2 above)
Comment (oliver-sanders) [reply tests/flakyfunctional/cylc-take-checkpoints/00-basic/flow.cylc]: I think the whole file is a rebase error?

Comment (wxtim) [reply tests/flakyfunctional/cylc-take-checkpoints/00-basic/flow.cylc]: No.

File: tests/flakyfunctional/cylc-take-checkpoints/00-basic/flow.cylc (hunk 5 above)
Comment (wxtim) [reply tests/flakyfunctional/cylc-take-checkpoints/00-basic/flow.cylc]: Yep - I see - and Have removed.

**Review by oliver-sanders — APPROVED** [review]

**Comment by oliver-sanders** [comment]
Could you possibly squash down those "fix" and "update" commits pre-merge?

**Comment by wxtim** [comment]
> Could you possibly squash down those "fix" and "update" commits pre-merge?

Hopefully that looks a bit better

**Pull request merged**